llvm/lib/Target/AArch64/AArch64LoadStoreOptimizer.cpp

```diff
@@ -965,6 +965,28 @@ static void updateDefinedRegisters(MachineInstr &MI, LiveRegUnits &Units,
       Units.addReg(MOP.getReg());
 }
 
+/// This function will add a new entry into the debugValueSubstitutions table
+/// when two instruction have been merged into a new one represented by \p
+/// MergedInstr.
+static void addDebugSubstitutionsToTable(MachineFunction *MF,
+                                         unsigned InstrNumToSet,
+                                         MachineInstr &OriginalInstr,
+                                         MachineInstr &MergedInstr) {
+
+  // Figure out the Operand Index of the destination register of the
+  // OriginalInstr in the new MergedInstr.
+  auto Reg = OriginalInstr.getOperand(0).getReg();
+  unsigned OperandNo = 0;
+  for (auto Op : MergedInstr.operands()) {
+    if (Op.getReg() == Reg)
+      break;
+    OperandNo++;
```

Comment:
Are we okay with this returning a bogus result if there is no Reg operand?

Reply:
I changed it so that we only make the substitution if we find the correct Reg Operand.